src/metabase/lib/drill_thru.cljc

```diff
@@ -55,9 +61,34 @@
           (assoc-in context [:column :lib/source-uuid] (lib.options/uuid ag))))
       context))
 
-(defmethod lib.metadata.calculation/display-info-method ::drill-thru
-  [query stage-number drill-thru]
-  (lib.drill-thru.common/drill-thru-info-method query stage-number drill-thru))
+;;; TODO: Missing drills: automatic insights, format.
+(def ^:private available-drill-thru-fns
+  [#'lib.drill-thru.distribution/distribution-drill
+   #'lib.drill-thru.column-filter/column-filter-drill
+   #'lib.drill-thru.foreign-key/foreign-key-drill
+   #'lib.drill-thru.object-details/object-detail-drill
+   #'lib.drill-thru.pivot/pivot-drill
+   #'lib.drill-thru.quick-filter/quick-filter-drill
+   #'lib.drill-thru.sort/sort-drill
+   #'lib.drill-thru.summarize-column/summarize-column-drill
+   #'lib.drill-thru.summarize-column-by-time/summarize-column-by-time-drill
+   #'lib.drill-thru.underlying-records/underlying-records-drill
+   #'lib.drill-thru.zoom-in-timeseries/zoom-in-timeseries-drill])
+
+(mu/defn ^:private all-contexts :- [:sequential {:min 1} ::lib.schema.drill-thru/context]
+  [query                             :- ::lib.schema/query
+   stage-number                      :- :int
+   {:keys [dimensions], :as context} :- ::lib.schema.drill-thru/context]
+  (if (empty? dimensions)
+    [context]
+    (let [stage            (lib.util/query-stage query stage-number)
+          returned-columns (lib.metadata.calculation/returned-columns query stage-number stage)]
+      (for [{:keys [column-name value]} dimensions
+            :let                        [col (lib.field/resolve-column-name-in-metadata column-name returned-columns)]
+            :when                       col]
+        (assoc context
+               :column col
+               :value value)))))
```

Comment:
I've got a better way to do this - the FE passes in real (JSON) columns for the dimensions, but the `metabase.lib.js/available-drill-thru` is throwing them away. it only extracts the column name because that was enough for the first drill that needed the dimensions. Underlying records needs fuller columns so I reworked the `lib.js` wrapper to pass them instead of just the name.

Your call - we can land this as is, and I can refactor it later, or I could split that part out of my changes and get it submitting.

Reply:
yeah, I think it might be easiest to land this one in first and you can follow on with your refactor, that way we have the tests that make sure zoom in timeseries drills come back and we know it still works after your refactor